### PR TITLE
implementing to publish key templates and blocking conditions into mu…

### PR DESCRIPTION
**Pull request by tharindu1st**
### Proposed changes in this pull request

This will addressed Publishing Blocking and Key Template Events into multiple traffic manager instances.

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

**Comment by tharindu1st** [comment]
resolves #https://github.com/wso2/product-apim/issues/2883

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java [hunk 1]
```diff
@@ -384,6 +384,16 @@
     public static final String OAUTH_RESPONSE_TOKEN_SCOPE = "scope";
     public static final String OAUTH_RESPONSE_EXPIRY_TIME = "expires_in";
     public static final String APP_DISPLAY_NAME = "DisplayName";
+    public static final String RECEIVER_URL = "receiverURL";
+    public static final String AUTHENTICATOR_URL = "authenticatorURL";
+    public static final String  USERNAME = "username";
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java]: Additional space

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/internal/APIManagerComponent.java [hunk 2]
```diff
@@ -222,7 +227,8 @@ protected void activate(ComponentContext componentContext) throws Exception {
             setupImagePermissions();
             APIMgtDBUtil.initialize();
 
-            configureJMSPublisher();
+            //configureJMSPublisher();
```
Comment (tgtshanika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/internal/APIManagerComponent.java]: Remove commented method

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 3]
```diff
@@ -4998,42 +4999,34 @@ private void publishBlockingEventUpdate(BlockConditionsDTO blockCondition) throw
...
                 .getAPIManagerConfigurationService().getAPIManagerConfiguration().getThrottleProperties();
 
         // Checking whether EventPublisherName is provided.  An empty HashMap is set so that it can be used to keep transport.jms
         // .Header value.
-        if (throttleProperties.getJmsPublisherParameters() != null && !throttleProperties.getJmsPublisherParameters()
-                .isEmpty()) {
-            eventAdapterService.publish(APIConstants.BLOCKING_EVENT_PUBLISHER, new HashMap<String, String>()
-                    , blockingMessage);
+        if (throttleProperties.getDataPublisher() != null && throttleProperties.getDataPublisher().isEnabled()) {
+            eventAdapterService.publish(APIConstants.BLOCKING_EVENT_PUBLISHER, Collections.EMPTY_MAP , blockingMessage);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Please format

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 4]
```diff
@@ -4998,42 +4999,34 @@ private void publishBlockingEventUpdate(BlockConditionsDTO blockCondition) throw
...
-        HashMap<String, String> keyTemplateMap = new HashMap<String, String>();
-        keyTemplateMap.put(APIConstants.POLICY_TEMPLATE_KEY, templateValue);
-        keyTemplateMap.put(APIConstants.TEMPLATE_KEY_STATE, state);
 
         ThrottleProperties throttleProperties = ServiceReferenceHolder.getInstance()
                 .getAPIManagerConfigurationService().getAPIManagerConfiguration().getThrottleProperties();
 
         // Getting JMS Publisher name from config. An empty HashMap is set so that it can be used to keep transport.jms
         // .Header value.
-        if (throttleProperties.getJmsPublisherParameters() != null && !throttleProperties.getJmsPublisherParameters()
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Need to remove/update the comment above.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 5]
```diff
@@ -4998,42 +4999,34 @@ private void publishBlockingEventUpdate(BlockConditionsDTO blockCondition) throw
...
     }
 
     private void publishKeyTemplateEvent(String templateValue, String state) {
-
+        OutputEventAdapterService eventAdapterService = ServiceReferenceHolder.getInstance().getOutputEventAdapterService();
+        Object[] objects = new Object[]{templateValue,state};
+        Event keyTemplateMessage = new Event(APIConstants.KEY_TEMPLATE_STREM_ID, System.currentTimeMillis(),
+                null, null, objects);
         //Publishing an event to notify that a policy has been added.
-        HashMap<String, String> keyTemplateMap = new HashMap<String, String>();
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Need to remove/update the comment above.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 6]
```diff
@@ -4998,42 +4999,34 @@ private void publishBlockingEventUpdate(BlockConditionsDTO blockCondition) throw
...
-        blockingMessage.put(APIConstants.BLOCKING_CONDITION_DOMAIN, tenantDomain);
+        Object[] objects = new Object[]{conditionType,conditionValue,state,tenantDomain};
+        Event blockingMessage = new Event(APIConstants.BLOCKING_CONDITIONS_STREAM_ID, System.currentTimeMillis(),
+                null, null, objects);
         ThrottleProperties throttleProperties = ServiceReferenceHolder.getInstance()
                 .getAPIManagerConfigurationService().getAPIManagerConfiguration().getThrottleProperties();
 
         // Checking whether EventPublisherName is provided.  An empty HashMap is set so that it can be used to keep transport.jms
         // .Header value.
-        if (throttleProperties.getJmsPublisherParameters() != null && !throttleProperties.getJmsPublisherParameters()
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Need to remove/update the comment above.

**Pull request merged**